File: 0101-0200/0165.Compare_Version_Numbers.py
#0ms





class Solution(object):
    def compareVersion(self, version1, version2):
        v1 = list(map(int, version1.split('.')))
        v2 = list(map(int, version2.split('.')))

        while len(v1) < len(v2):
            v1.append(0)
        while len(v2) < len(v1):
            v2.append(0)

        for a, b in zip(v1, v2):
            if a < b:
                return -1
            elif a > b:
                return 1

        return 0





#0ms





class Solution:
  def compareVersion(self, version1: str, version2: str) -> int:
    arr1 = version1.split(".") 
    arr2 = version2.split(".") 
    n = len(arr1)
    m = len(arr2)
    arr1 = [int(i) for i in arr1]
    arr2 = [int(i) for i in arr2]
    if n>m:
      for i in range(m, n):
         arr2.append(0)
    elif m>n:
      for i in range(n, m):
         arr1.append(0)
    
    for i in range(len(arr1)):
      if arr1[i]>arr2[i]:
         return 1
      elif arr2[i]>arr1[i]:
         return -1
    return 0

File: 0101-0200/0165.Compare_Version_Numbers.rs
//0ms





use std::iter::{repeat, zip};

impl Solution {
    pub fn compare_version(v1: String, v2: String) -> i32 {
        let n = v1.len().abs_diff(v2.len());
        let split_parse = |ver: String| -> Vec<i32> {
            ver.split('.')
                .filter_map(|s| s.parse::<i32>().ok())
                .chain(repeat(0).take(n))
                .collect()
        };

        zip(split_parse(v1), split_parse(v2))
            .skip_while(|(a, b)| a == b)
            .map(|(a, b)| if a > b { 1 } else { -1 })
            .next()
            .unwrap_or(0)
    }
}






//0ms





impl Solution {
    pub fn compare_version(version1: String, version2: String) -> i32 {

        let mut iter1 = version1.split('.').map(|s| s.parse::<i32>().unwrap());
        let mut iter2 = version2.split('.').map(|s| s.parse::<i32>().unwrap());

        loop {
            match (iter1.next(), iter2.next()) {
                (Some(r1), Some(r2)) => {
                    if r1 > r2 {
                        break 1;
                    } else if r1 < r2 {
                        break -1;
                    }
                }
                (Some(r1), None) => if r1 > 0 { break 1; },
                (None, Some(r2)) => if r2 > 0 { break -1; },
                (None, None) => break 0,
            }
        }
    }
}
